core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -3560,7 +3560,12 @@ protected boolean checkUserNameValid(String userName) throws UserStoreException
         if (MultitenantUtils.isEmailUserName()) {
             regularExpression = realmConfig
                     .getUserStoreProperty(UserCoreConstants.RealmConfig.PROPERTY_USER_NAME_WITH_EMAIL_JS_REG_EX);
-            if (regularExpression == null) {
+
+            if (StringUtils.isEmpty(regularExpression) || StringUtils.isEmpty(regularExpression.trim())) {
+                regularExpression = realmConfig.getUserStoreProperty(UserCoreConstants.RealmConfig
+                        .PROPERTY_USER_NAME_JAVA_REG_EX);
```

Comment:
If we are going with PROPERTY_USER_NAME_JAVA_REG_EX better change the relevant docs accordingly.
Especially Email Username Configuring Documentation [1].
[1] https://docs.wso2.com/display/IS540/Using+Email+Address+as+the+Username

Reply:
Noted.